rtabmap_odom/src/OdometryROS.cpp

```diff
@@ -543,10 +544,45 @@ void OdometryROS::mainLoop()
 	Transform groundTruth;
 	if(!data.imageRaw().empty() || !data.laserScanRaw().isEmpty())
 	{
-		if(previousStamp_>0.0 && previousStamp_ >= header.stamp.toSec())
+		// Detect time jump in the past
+		double clockNow = ros::Time::now().toSec();
+		if(previousClockTime_>0.0 && previousClockTime_ > clockNow)
+		{
+			NODELET_WARN("Odometry: Detected jump back in time of %f sec. Odometry is "
+				"automatically reset to latest computed pose!",
+				previousClockTime_ - clockNow);
+			SensorData dataCpy = dataToProcess_;
+			std_msgs::Header headerCpy = dataHeaderToProcess_;
+			double previousCpy = previousClockTime_;
+			this->reset(odometry_->getPose());
+			if(previousCpy > headerCpy.stamp.toSec()) {
+				// new frame is using new clock, process it now
+				dataToProcess_ = dataCpy;
+				dataHeaderToProcess_ = headerCpy;
+				dataReady_.release();
+				NODELET_WARN("Odometry: Restarting with frame: %f (clock previous=%f, new=%f)",
+					headerCpy.stamp.toSec(), previousCpy, clockNow);
+			}
+			else {
+				// skip that old frame
+				NODELET_WARN("Odometry: skipping frame: %f (clock previous=%f, new=%f)",
+					headerCpy.stamp.toSec(), previousCpy, clockNow);
+			}
+			previousClockTime_ = clockNow;
+			return;
+		}
+		previousClockTime_ = clockNow;
+
+		if(header.stamp.toSec() > clockNow) {
```

Comment:
We cannot do that, otherwise multi-computer mapping won't work (in case the mapping computer's clock is slightly in the past).